pennylane/ops/qubit.py

```diff
@@ -897,6 +916,10 @@ def _matrix(cls, *params):
                 " Allowed characters are I, X, Y and Z".format(pauli_word)
             )
 
+        # Simplest case is if the Pauli is the identity matrix
+        if pauli_word == "I" * len(pauli_word):
+            return np.exp(-1j * theta / 2) * np.eye(2 ** len(pauli_word))
```

Comment:
Just curious: could it make some users think that this would have a contribution in the circuit? (e.g. introducing noise on HW)

Reply:
Yes actually I wasn't quite sure how to treat this one. Technically there is a global phase in the matrix, if you directly evaluate e^{-itP}, however it doesn't really matter in the end.